Adapt links to cantonal geoportals

nadine-piveteau opened the pull request:
See[ issue #4020](https://github.com/geoadmin/mf-geoadmin3/issues/4020#issuecomment-341657468)

loicgasser commented on chsdi/templates/htmlpopup/cadastralwebmap.mako:
```diff
@@ -10,7 +10,7 @@
   from chsdi.lib.validation.identify import IdentifyServiceValidation
   request = context.get('request')
-  grid = getTileGrid(21781)()
+  grid = getTileGrid(2056)()
```
pass in params.srid

loicgasser commented on chsdi/templates/htmlpopup/cadastralwebmap.mako:
```diff
@@ -22,12 +22,14 @@
                   (c['bbox'][1] + c['bbox'][3]) / 2]
   clickCoord = request.params.get('coord').split(',') if request.params.get('coord') else defaultCoord
+  c['clickCoordLv03'] = [clickCoord[0][1:],
+                         clickCoord[1][1:]]
```
why do you use [1:] ? coords in this case is also a list of 2 coordinates which srid is determined by `params.srid`
so make sure you clearly have both `clickCoordLv03` and `clickCoordLv95` in the Mako context because `clickCoord` could be either LV95 or LV03.

nadine-piveteau replied:
clickCoord returns a tuple of strings which represents the coordinates in LV95. As I needed the coordinates in LV03 I removed the 2's (east) and the 1's (north) by shortening the string by its first character. I agree, it's not a very nice hack :D 
Anyway, I will in a next step add it to the context variable such that it is not necessary anymore to apply that .

loicgasser commented on chsdi/templates/htmlpopup/cadastralwebmap.mako:
```diff
@@ -22,12 +22,14 @@
-  c['bboxlv95'] = [c['bbox'][0] + 2000000, c['bbox'][1] + 1000000,
-                   c['bbox'][2] + 2000000, c['bbox'][3] + 1000000]
+  c['bboxlv03'] = [c['bbox'][0] - 2000000, c['bbox'][1] - 1000000,
+                   c['bbox'][2] - 2000000, c['bbox'][3] - 1000000]
```
you could simply replace bbox with `bboxlv95` and `bboxlv03`, thus passing explicitly the correct values in the mako.

loicgasser replied:
FYI: this method `shift_to(coords, srid)` will transform both point and bbox coordinates to the correct value.

https://github.com/geoadmin/mf-chsdi3/blob/master/chsdi/lib/helpers.py#L316

nadine-piveteau commented:
Thx for the feedback! I added the requested modifications. Because of these changes I also had to modify the code for OpenData-AV. Thus, I would be grateful if you could test both, CadastralWebMap and OpenData-AV. Thx in advance!
(There are two tests concerning Sphinx and the search service that failed. I think it has nothing to do with my modifications, but let me know, if I should fix that!)
 
[Testlink](https://mf-geoadmin3.dev.bgdi.ch/ltpin/src/?api_url=%2F%2Fmf-chsdi3.int.bgdi.ch%2Fltpin_cadastralwebmap&lang=de&topic=ech&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.kantone.cadastralwebmap-farbe,ch.swisstopo-vd.amtliche-vermessung&E=2692010.84&N=1195546.62&zoom=2)

@AFoletti : I've also updated the link to SG-Geoportal.

nadine-piveteau commented:
I just realized that some PDF-Download links do not work anymore given that some cantons expect LV03 coordinates. Working on that...

loicgasser commented:
ok thanks, it's a bit of a juggle between lv95 and lv03 for cantons as well.

loicgasser commented:
sphinx is because of some on going devs on search server. (missing quadindex for locations_preview)
so unrelated

loicgasser commented on chsdi/templates/htmlpopup/cadastralwebmap_opendata.mako:
```diff
@@ -12,35 +12,49 @@
           self.mapExtent = request.params.get('mapExtent', defaultExtent)
           self.imageDisplay = request.params.get('imageDisplay', defaultImageDisplay)
+          self.coord = request.params.get('coord')
+  toto
```
toto?

loicgasser commented on chsdi/templates/htmlpopup/cadastralwebmap.mako:
```diff
@@ -9,25 +9,30 @@
-  c['bboxlv95'] = [c['bbox'][0] + 2000000, c['bbox'][1] + 1000000,
-                   c['bbox'][2] + 2000000, c['bbox'][3] + 1000000]
+  params = CadastralWebMapParams(request)
+  c['bboxlv95'] =  list(params.mapExtent.bounds)
```
if sr parameter == 2056 then this is incorrect...
why did you remove the if we added yesterday?

loicgasser replied:
21781 I mean

loicgasser replied:
and you need the fix in geoadmin otherwise it won't work.

nadine-piveteau replied:
Yes I removed it because I noticed that the params coord and mapExtent are delivered per default in Lv95 also when sr=21781.

loicgasser replied:
no, incorrect default is lv03 still, you have to pass in sr parameter....
If you tested with prod it's normal it didn't work.

loicgasser commented on chsdi/templates/htmlpopup/cadastralwebmap_opendata.mako:
```diff
@@ -12,35 +12,49 @@
   params = CadastralWebMapParams(request)
-  c['bbox'] = params.mapExtent.bounds
-  c['bboxlv95'] = [2000000 + c['bbox'][0], 1000000 + c['bbox'][1], 2000000 + c['bbox'][2], 1000000 + c['bbox'][3]]
+  c['bboxlv95'] = list(params.mapExtent.bounds)
```
same here:
```python
if params.srid == 2056:
    c['bboxlv95'] = list(params.mapExtent.bounds)
    c['bboxlv03'] = shift_to(c['bboxlv95'], 21781)
else:
    c['bboxlv03'] = list(params.mapExtent.bounds)
    c['bboxlv95'] = shift_to(c['bboxlv03'], 2056)
```

loicgasser commented on chsdi/templates/htmlpopup/cadastralwebmap_opendata.mako:
```diff
@@ -12,35 +12,49 @@
+
+  defaultCoord = [(c['bboxlv95'][0]+c['bboxlv95'][2])/2,
+                  (c['bboxlv95'][1]+c['bboxlv95'][3])/2]
+  c['clickCoordLv95'] = [float(a) for a in params.coord.split(',')] if params.coord else defaultCoord
```
here you need to check srid as well.

loicgasser replied:
ah no that's ok

loicgasser commented on chsdi/templates/htmlpopup/cadastralwebmap_opendata.mako:
```diff
@@ -12,35 +12,49 @@
+    lon = c['clickCoordLv95'][1]
+  else:
+    lat = c['clickCoordLv03'][0]
+    lon = c['clickCoordLv03'][1]
```
this should always be lv03...

nadine-piveteau commented:
yes I redeployed the branch to int.

loicgasser commented:
thanks!
